File: src/main/java/src/SistemaRPG/Guerrero.java
package SistemaRPG;

public class Guerrero extends Personaje {
    private int fuerza;

    public Guerrero(String nombre, int puntosVida, int fuerza) {
        super(nombre, puntosVida);
        this.fuerza = fuerza;
    }

    @Override
    public String toString() {
        return super.toString() + " (Guerrero, Fuerza: " + fuerza + ")";
    }

    @Override
    public void atacar(Personaje objetivo) {
        int danio = fuerza;
        objetivo.recibirDanio(danio);
        System.out.println(getNombre() + " ataca a " + objetivo.getNombre() + " y causa " + danio + " de daño.");
    }
}

File: src/main/kotlin/SistemaRPG/Guerrero.kt
package SistemaRPG

class Guerrero(nombre: String, puntosVida: Int, private val fuerza: Int) : Personaje(nombre, puntosVida) {
    override fun toString(): String {
        return super.toString() + " (Guerrero, Fuerza: " + fuerza + ")"
    }

    override fun atacar(objetivo: Personaje?) {
        val danio = fuerza
        objetivo?.recibirDanio(danio)
        if (objetivo != null) {
            println(nombre + " ataca a " + objetivo.nombre + " y causa " + danio + " de daño.")
        }
    }
}